foreign-key-mapping/README.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: pattern
+title: Foreign-Key-Mapping
+folder: foreign-key-mapping
+permalink: /patterns/foreign-key-mapping/
+categories: Object-Relational Structural
+tags:
+- foreign-key
+---
+
+## Intent
+Maps an association between objects to a foreign key reference between tables.
+
+Key points:
+
+- A Foreign Key Mapping maps an object reference to a foreign key in the database.
+
+- In java programming, objects can refer to each other directly by object references.
+
+- In a relational database, a FOREIGN KEY is a key used to link two tables together.
+
+- In a relational database, a FOREIGN KEY is a field (or collection of fields) in one table that refers to the PRIMARY KEY in another table.
+
+- The table containing the foreign key is called the child table, and the table containing the candidate key is called the referenced or parent table.
```

Comment:
I suggest we put the key points under separate header. The available headers are listed in https://github.com/iluwatar/java-design-patterns/wiki/02.-Pattern-template.